src/tests/class_tests/openms/source/MRMFeatureSelector_test.cpp

```diff
@@ -0,0 +1,357 @@
+  MRMFeatureSelector_test selector;
+  TEST_STRING_EQUAL(selector.remove_spaces("h e ll o"), "hello");
+  TEST_STRING_EQUAL(selector.remove_spaces("hello"), "hello");
+  TEST_STRING_EQUAL(selector.remove_spaces(""), "");
+  TEST_STRING_EQUAL(selector.remove_spaces("A    B"), "AB");
+}
+END_SECTION
+
+START_SECTION(constructToList())
+{
+  MRMFeatureSelector_test selector;
+  FeatureMap feature_map;
+  FeatureXMLFile feature_file;
+  feature_file.load(features_path, feature_map);
+
+  vector<pair<double, String>> time_to_name;
+  map<String, vector<Feature>> feature_name_map;
+  selector.setSelectTransitionGroup("true");
+  selector.constructToList(feature_map, time_to_name, feature_name_map);
+
+  TEST_EQUAL(time_to_name.size(), 117)
+  TEST_EQUAL(feature_name_map.size(), 117)
+
+  sort(time_to_name.begin(), time_to_name.end());
+
+  pair<double, String> *p = nullptr;
+
+  p = &time_to_name.front();
+  TEST_REAL_SIMILAR(p->first, 0)
+  TEST_STRING_EQUAL(p->second, "arg-L")
+
+  p = &time_to_name[1];
+  TEST_REAL_SIMILAR(p->first, 0.167913821)
+  TEST_STRING_EQUAL(p->second, "orn")
+
+  p = &time_to_name[54];
+  TEST_REAL_SIMILAR(p->first, 61.76161499)
+  TEST_STRING_EQUAL(p->second, "35cgmp")
+
+  p = &time_to_name[99];
+  TEST_REAL_SIMILAR(p->first, 92.88219725)
+  TEST_STRING_EQUAL(p->second, "itp")
+
+  p = &time_to_name.back();
+  TEST_REAL_SIMILAR(p->first, 99.98770892)
+  TEST_STRING_EQUAL(p->second, "succoa")
+}
+END_SECTION
+
+START_SECTION(weight_func())
+{
+  MRMFeatureSelector_test selector;
+  double score = -1.0;
+
+  score = selector.weight_func(3413.0, "lambda score: score*1.0");
+  TEST_REAL_SIMILAR(score, 3413.0)
+  score = selector.weight_func(341.0, "lambda score: 1/score");
+  TEST_REAL_SIMILAR(score, 0.002932551)
+  score = selector.weight_func(341.0, "lambda score: log(score)");
+  TEST_REAL_SIMILAR(score, 5.831882477)
+  score = selector.weight_func(96640.0, "lambda score: 1/log(score)");
+  TEST_REAL_SIMILAR(score, 0.087117)
+  score = selector.weight_func(341.0, "lambda score: 1/log10(score)");
+  TEST_REAL_SIMILAR(score, 0.394827074)
+}
+END_SECTION
+
+START_SECTION(compute_score())
+{
+  MRMFeatureSelector_test selector;
+  double score;
+  Feature feature;
+  feature.setMetaValue("sn_ratio", 6.84619503982874);
+  feature.setMetaValue("peak_apices_sum", 96640.0);
+
+  selector.setScoreWeights({{"sn_ratio", "lambda score: 1/log(score)"}});
+  score = selector.compute_score(feature);
+  TEST_REAL_SIMILAR(score, 0.5198334582314795)
+
+  selector.setScoreWeights({{"peak_apices_sum", "lambda score: 1/log10(score)"}});
+  score = selector.compute_score(feature);
+  TEST_REAL_SIMILAR(score, 0.20059549093267626)
+
+  selector.setScoreWeights({{"sn_ratio", "lambda score: 1/log(score)"}, {"peak_apices_sum", "lambda score: 1/log10(score)"}});
+  score = selector.compute_score(feature);
+  TEST_REAL_SIMILAR(score, 0.10427624775717449)
+}
+END_SECTION
+
+// START_SECTION(schedule_MRMFeaturesQMIP() integer) // integer variable type
```

Comment:
I would recommend to leave this in there.

Reply:
Meaning, leave it as a commented-out piece of code?
The reason I commented it out is the time it takes to process (being in the order of minutes).

Reply:
I see.  OK just leave it as a comment then.